components/public-api/gitpod/v1/workspace.proto

```diff
@@ -3,25 +3,119 @@ syntax = "proto3";
+  // Deleted workspaces cannot be started again.
+  rpc DeleteWorkspace(DeleteWorkspaceRequest)
+      returns (DeleteWorkspaceResponse) {}
+
+  // UpdateWorkspace updates the meta data of a workspace.
+  rpc UpdateWorkspace(UpdateWorkspaceRequest)
+      returns (UpdateWorkspaceResponse) {}
+
+  // ListSupportedWorkspaceClasses enumerates all available workspace classes.
+  rpc ListSupportedWorkspaceClasses(ListSupportedWorkspaceClassesRequest)
+      returns (ListSupportedWorkspaceClassesResponse) {}
+
+  // GetDefaultWorkspaceImage returns the default workspace image of specified
+  // workspace.
+  rpc GetDefaultWorkspaceImage(GetDefaultWorkspaceImageRequest)
+      returns (GetDefaultWorkspaceImageResponse) {}
+
+  // SendHeartBeat sends a heartbeat to activate the workspace
+  rpc SendHeartBeat(SendHeartBeatRequest) returns (SendHeartBeatResponse) {}
+
+  // GetWorkspaceOwnerToken returns an owner token of workspace.
+  rpc GetWorkspaceOwnerToken(GetWorkspaceOwnerTokenRequest)
+      returns (GetWorkspaceOwnerTokenResponse) {}
+
+  // GetWorkspaceEditorCredentials returns an credentials that is used in editor
+  // to encrypt and decrypt secrets
+  rpc GetWorkspaceEditorCredentials(GetWorkspaceEditorCredentialsRequest)
+      returns (GetWorkspaceEditorCredentialsResponse) {}
+
+  // WatchWorkspaceImageBuildLogs watches the image build logs of prebuild
+  // workspace.
+  rpc WatchWorkspaceImageBuildLogs(WatchWorkspaceImageBuildLogsRequest)
+      returns (stream WatchWorkspaceImageBuildLogsResponse) {}
+
+  // ListWorkspacePorts lists workspace ports.
+  rpc ListWorkspacePorts(ListWorkspacePortsRequest)
+      returns (ListWorkspacePortsResponse) {}
+
+  // UpdateWorkspacePort updates a workspace port.
+  rpc UpdateWorkspacePort(UpdateWorkspacePortRequest)
+      returns (UpdateWorkspacePortResponse) {}
+
+  // CreateWorkspaceSnapshot creates a snapshot of the workspace that can be
+  // shared with others.
+  rpc CreateWorkspaceSnapshot(CreateWorkspaceSnapshotRequest)
+      returns (CreateWorkspaceSnapshotResponse) {}
+
+  // WaitWorkspaceSnapshot waits for the snapshot to be available or failed.
+  rpc WaitWorkspaceSnapshot(WaitWorkspaceSnapshotRequest)
+      returns (WaitWorkspaceSnapshotResponse) {}
+
+  // ListWorkspaceSnapshots lists the snapshots.
+  rpc ListWorkspaceSnapshots(ListWorkspaceSnapshotsRequest)
+      returns (ListWorkspaceSnapshotsResponse) {}
+
+  // ListWorkspaceInstances lists the instances of a workspace.
+  rpc ListWorkspaceInstances(ListWorkspaceInstancesRequest)
+      returns (ListWorkspaceInstancesResponse) {}
+
+  // RestoreWorkspace restores a workspace.
+  rpc RestoreWorkspace(RestoreWorkspaceRequest)
+      returns (RestoreWorkspaceResponse) {}
+}
+
+message GetWorkspaceRequest {
+  reserved "id";
+  reserved 1;
+
+  WorkspaceScope scope = 2;
+
+  string workspace_id = 3;
 }
 
-message GetWorkspaceRequest { string id = 1; }
+message GetWorkspaceResponse {
+  reserved "item";
```

Comment:
we don't need it, we can break now, since now one is actually using gitpod/v1 in production

Reply:
When generating SDKs, it will run breaking change detect. So I can just disable it?